### PR TITLE
docs: add document for styled-components

### PR DESCRIPTION
**Pull request by Asuka109**

### CODE REVIEW

**Review by h-a-n-a — APPROVED** [review]
Merge until new version has been released.

File: docs/en/guide/loader.mdx [hunk 1]
```diff
@@ -225,13 +225,80 @@ export type RspackExperiments = {
...
-The configuration can be found in the [builtins](/config/builtins) section:
+Some fields can be found in the [builtins](/config/builtins) section:
 
 - [import](/config/builtins#builtinspluginimport)
 - [emotion](/config/builtins#builtinsemotion)
 - [relay](/config/builtins#builtinsrelay)
+
+And some new features will only provided by `builtin:swc-loader`.
+
+##### options.rspackExperiments.styledComponents
```
Comment (h-a-n-a) [inline docs/en/guide/loader.mdx:242]: Would you please add a `ApiMeta` label for `addedVersion`?

File: docs/zh/guide/loader.mdx [hunk 2]
```diff
@@ -226,13 +226,80 @@ export type RspackExperiments = {
...
-这些配置项可以在 [builtins](/config/builtins) 下找到：
+其中部分配置项可以在 [builtins](/config/builtins) 下找到：
 
 - [import](/config/builtins#builtinspluginimport)
 - [emotion](/config/builtins#builtinsemotion)
 - [relay](/config/builtins#builtinsrelay)
+
+此外一些新的特性将仅通过 `builtin:swc-loader` 提供。
+
+##### options.rspackExperiments.styledComponents
```
Comment (h-a-n-a) [inline docs/zh/guide/loader.mdx:243]: Same to this one

Comment (Asuka109) [reply docs/zh/guide/loader.mdx:243]: okk

**Review by h-a-n-a — APPROVED** [review]

**Pull request merged**